src/PhpSpec/CodeGenerator/Generator/PromptingGenerator.php

```diff
@@ -148,9 +159,23 @@ private function createDirectoryIfItDoesExist($filepath)
      */
     private function generateFileAndRenderTemplate(ResourceInterface $resource, $filepath)
     {
+        $fileExists = $this->fileAlreadyExists($filepath);
         $content = $this->renderTemplate($resource, $filepath);
 
         $this->filesystem->putFileContents($filepath, $content);
         $this->io->writeln($this->getGeneratedMessage($resource, $filepath));
+
+        if (!$fileExists) {
```

Comment:
If the file exists and the user chooses to overwrite it then it would be a change, not a file creation. There could potentially be an event dispatched when an existing file changes if a use case arises.